src/hera/workflows/script.py

```diff
@@ -760,6 +755,27 @@ def script_wrapper(func: Callable[FuncIns, FuncR]) -> Callable:
         def task_wrapper(*args, **kwargs) -> Union[FuncR, Step, Task, None]:
             """Invokes a `Script` object's `__call__` method using the given SubNode (Step or Task) args/kwargs."""
             if _context.active:
+                if len(args) == 1 and isinstance(args[0], (InputV1, InputV2)):
+                    arguments = args[0]._get_as_arguments()
+                    arguments_list = [
+                        *(arguments.artifacts or []),
+                        *(arguments.parameters or []),
+                    ]
+
+                    subnode = s.__call__(arguments=arguments_list, **kwargs)
+                    if not subnode:
+                        raise SyntaxError("Cannot use Pydantic I/O outside of a DAG, Steps or Parallel context")
+
+                    output_class = inspect.signature(func).return_annotation
+                    if not output_class or output_class is NoneType:
+                        return None
+
+                    if not issubclass(output_class, (OutputV1, OutputV2)):
+                        raise SyntaxError("Cannot use Pydantic input type without a Pydantic output type")
```

Comment:
it should be better to check these in tests.
seems to be submission tests?

Reply:
Added tests